File: AdvancePython/Day-3 Advanced OOP concepts/Task-5.py
# Task 5:
class Calculator:

    @staticmethod
    def add(a: int, b: int) -> int:
        return a + b

    @staticmethod
    def sub(a: int, b: int) -> int:
        return a - b

    @staticmethod
    def mul(a: int, b: int) -> int:
        return a * b

    @staticmethod
    def div(a: int, b: int) -> float:
        assert b != 0, "Value cannot divide by zero"  # only the divisor cannot be 0
        return a / b


print(f"Addition: {Calculator.add(10, 5)}")
print(f"Subtraction: {Calculator.sub(10, 5)}")
print(f"Multiplication: {Calculator.mul(10, 5)}")
print(f"Division: {Calculator.div(10, 5)}")
print(f"Division: {Calculator.div(10, 0)}")

File: node.js/Day-3/Task-5.js
// Task 5:

class Calculator {
    constructor(a, b) {
        this.a = a;
        this.b = b;
    }

    add() {
        console.log(`${this.a} + ${this.b} =`, this.a + this.b);
    }

    sub() {
        console.log(`${this.a} - ${this.b} =`, this.a - this.b);
    }

    mul() {
        console.log(`${this.a} * ${this.b} =`, this.a * this.b);
    }

    div() {
        console.log(`${this.a} / ${this.b} =`, this.a / this.b);
    }
}

const calc = new Calculator(423, 64);
calc.add();
calc.sub();
calc.mul();
calc.div();